package/hunter/HunterInit.cmake

```diff
@@ -0,0 +1,12 @@
+file(
+  DOWNLOAD
+  https://raw.githubusercontent.com/hunter-packages/gate/master/cmake/HunterGate.cmake
+  ${CMAKE_CURRENT_LIST_DIR}//HunterGate.cmake
+)
```

Comment:
I'm not sure what the exact behavior of CMake's `file(DOWNLOAD)` (could not find any concrete info on it) and I have a bad feeling that it could attempt a download on every CMake run. Could this be wrapped in `if(NOT EXISTS  ${CMAKE_CURRENT_LIST_DIR}/HunterGate.cmake)`?

The other option I mentioned in #297 was putting the file into the toolchains repository. But that would require the user to explicitly init the toolchains submodule, which is not needed for anything else. So I think this is a better solution :+1: However, to prevent potential backwards-incompatible changes, could we pin it to some concrete revision? Current master is `https://raw.githubusercontent.com/hunter-packages/gate/c4e5b3e4d1c97fb46ae86af7f59baac95f7f1cf0/cmake/HunterGate.cmake`.

Reply:
> Could this be wrapped in if(NOT EXISTS ${CMAKE_CURRENT_LIST_DIR}/HunterGate.cmake)?

This is done

> The other option I mentioned in #297 was putting the file into the toolchains repository. But that would require the user to explicitly init the toolchains submodule, which is not needed for anything else. So I think this is a better solution 👍

I thought so too! And it was the only solution since hunter uses `git archive`during its build (and we of course lose the submodules this way...)

I pushed a commit with these modifications, and tested it inside a travis like container, like so; so that I am confident about it.

````bash
> cd /path/to/magnum_hunter/Docker/ubuntu
# Inside the docker container 
> /sources_docker >  rm magnum/package/hunter/HunterGate.cmake
> /sources_docker >  ./TLDR_hunter.py test-build magnum gcc-7-cxx14 --clean
> file magnum/package/hunter/HunterGate.cmake
magnum/package/hunter/HunterGate.cmake: ASCII text
````